### PR TITLE
Fixes smugglers satchels being able to refresh items forever

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Items spawned in satchels may no longer be placed into a satchel once removed.

This means you can't store a used holoparasite injector in round A, have it spawn as a fresh one in round B, then use and replace it again to use in round C (and then in D and E and F and so on forever)

### CODE REVIEW

**Comment by Shadowflame909** [comment]
We shouldn't of found out about the dead gimmick. WHY SLIME PERSON KOR, WHYYYYYYYYYYYYYYYYY

File: code/controllers/subsystem/persistence.dm [hunk 1]
```diff
@@ -62,7 +63,8 @@ SUBSYSTEM_DEF(persistence)
 		if(isfloorturf(F.loc) && !isplatingturf(F.loc))
 			F.hide(1)
 		if(ispath(path))
-			new path(F)
+			var/spawned_item = new path(F)
+			spawned_objects += spawned_item
```
Comment (lzimann) [inline code/controllers/subsystem/persistence.dm]: make it `spawned_objects[spawned_item] = TRUE`

File: code/game/objects/items/storage/backpack.dm [hunk 2]
```diff
@@ -326,6 +326,12 @@
 			new R(src)
 		revealed = 1
 
+/obj/item/storage/backpack/satchel/flat/can_be_inserted(obj/item/W, stop_messages = 0, mob/user)
+	if(W in SSpersistence.spawned_objects)
```
Comment (lzimann) [inline code/game/objects/items/storage/backpack.dm]: and `SSpersistence.spawned_objects[W]` here

File: code/game/objects/items/storage/backpack.dm [hunk 3]
```diff
@@ -326,6 +326,12 @@
 			new R(src)
 		revealed = 1
 
+/obj/item/storage/backpack/satchel/flat/can_be_inserted(obj/item/W, stop_messages = 0, mob/user)
+	if(W in SSpersistence.spawned_objects)
+		to_chat(user, "<span class='warning'>[W] is unstable after its journey through space and time, it wouldn't survive another trip.</span>")
+		return 0
```
Comment (lzimann) [inline code/game/objects/items/storage/backpack.dm]: `FALSE`

File: code/game/objects/items/storage/backpack.dm [hunk 4]
```diff
@@ -326,6 +326,12 @@
 			new R(src)
 		revealed = 1
 
+/obj/item/storage/backpack/satchel/flat/can_be_inserted(obj/item/W, stop_messages = 0, mob/user)
+	if(W in SSpersistence.spawned_objects)
+		to_chat(user, "<span class='warning'>[W] is unstable after its journey through space and time, it wouldn't survive another trip.</span>")
+		return 0
+	..()
```
Comment (lzimann) [inline code/game/objects/items/storage/backpack.dm]: `return ..()`

**Comment by KorPhaeron** [comment]
@Shadowflame909 It got to the point where an admin decided to mass delete half the persistence pool, would rather balance satchels to stop the calls to remove them entirely

**Comment by Shadowflame909** [comment]
Oh. Well then thank you for keeping another gimmick alive. Instead of removing another feature. coderbus hates fun. x.x

**Comment by Shadowflame909** [comment]
The more people find out about dead things. The more things get removed. Whats next coderbus, will service borgs lose there spraycan that runs on there battery? WILL THEY

**Comment by Wilchenx** [comment]
@Shadowflame909 so you think getting holoparasites every round is okay?


😕

**Comment by Shadowflame909** [comment]
@Wilchenx  No. Thats problematic. But my argument is that they dont get them every round, as they only have a slight chance for a smuggler satchel to re-spawn. And they may not even be in the satchel. I think a better fix instead of removing things we love is to either, make it so you cant get a smuggler satchel and a holoparasite injector. Maybe by not allowing the parasite injector to be put into one. Or, maybe make it harder to find the smuggler satchel? Removing things is a bad thing. I say Revamp and keep or leave it alone.

**Comment by KorPhaeron** [comment]
It's actually a fairly high chance for them to spawn since using one didn't remove it from the pool. 1/3 the list of satchels on basil was holoparasite bags

This way you can still save holoparasites but you can only retrieve them once after that, which will regulate the pool instead of letting people flood it

**Comment by Shadowflame909** [comment]
I guess I can see how a traitor could do it alot, well. As long as smuggler satchels aren't getting removed because someone from coderbus just now looked at it and thought "oh fuck remove I have never seen this before" But, What kind of traitor would go every traitor round just spawning smuggler satchels and holoparasite injectors??? Now what do I gotta do to get them re-added to null crates..

**Comment by ExcessiveUseOfCobblestone** [comment]
smuggler satchels are always in the game and not necessarily traitor bought.

They randomly spawn under tiles.

**Comment by Shadowflame909** [comment]
Are you sure it wasnt the work of traitors from 2009?

**Comment by arsserpentarium** [comment]
Maybe fix renovation of items instead of restriction to put them in satchels?

**Comment by Iatots** [comment]
I have never, ever found anything that wasn't a crowbar and tile. What am I doing wrong?

**Comment by KorPhaeron** [comment]
@iatots you gotta personally flood the pool in the same spot every round until the odds are on your side

File: code/game/objects/items/storage/backpack.dm [hunk 5]
```diff
@@ -326,6 +326,12 @@
 			new R(src)
 		revealed = 1
 
+/obj/item/storage/backpack/satchel/flat/can_be_inserted(obj/item/W, stop_messages = 0, mob/user)
+	if(SSpersistence.spawned_objects[W])
+		to_chat(user, "<span class='warning'>[W] is unstable after its journey through space and time, it wouldn't survive another trip.</span>")
+		return 0
+	..()
```
Comment (lzimann) [inline code/game/objects/items/storage/backpack.dm]: forgot this bit, `return ..()` and `return FALSE`

**Comment by Incoming5643** [comment]
I'm surprised someone cared enough to abuse this. I'm glad honestly to see that the system still works properly.

Anyway there's an alternate solution to this. You can set the item in spawned_object to set its persistence_replacement to a booby prize item path if it's removed from the bag. Then if they try to cheese it to a future it will instead be that booby prize item in future rounds.

**Comment by KorPhaeron** [comment]
Might be confusing and frustrating to people who found a satchel without trying to cheese the whole thing though

**Comment by AlexTheSysop42** [comment]
The problem is, that prevents someone from getting a non-consumeable item (e.g. energy gun or emag) in two different rounds

**Comment by ShiggyDiggyDo** [comment]
It's tiring to see people having holoparasites every single goddamn round

**Comment by Xhuis** [comment]
There was no changelog for this.

**Pull request merged**